backend/app.py

```diff
@@ -32,12 +33,17 @@ def generate_images_api():
     
     if config.IMAGE_MODEL == "potato":
         time.sleep(5)
+        seed = None
         generated_images = utils.separate_grid(Image.open('./static/placeholder.jpeg'))
     else:
-        generated_images = image_model.generate_images(translated_prompt)
+        generated_images, seed = image_model.generate_images(translated_prompt)
 
+    nsfw_image = None
     if config.FILTER_IMAGES:
-        generated_images = util_nsfwchecks.filter_images(generated_images, config.NSFW_TRESHOLD)
+        generated_images, nsfw_image = util_nsfwchecks.filter_images(generated_images, config.NSFW_TRESHOLD)
+
+    if config.USE_DATABASE:
+        database.save_prompt(db_connection, translated_prompt, seed, profane, nsfw_image)
```

Comment:
I think it would be nice to save down the original untranslated prompt as well

Reply:
Added this.

Reply:
Also added the language.